torch/csrc/jit/serialization/export.cpp

```diff
@@ -833,32 +833,26 @@ unsigned long long int GraphEncoder::GetGraphProtoSize(
     onnx::GraphProto* graph_proto,
     const std::shared_ptr<Graph>& graph,
     const std::map<std::string, at::Tensor>& initializers) {
-  unsigned long long int sizes = 0;
+  onnx::GraphProto graph_proto_copy = onnx::GraphProto(*graph_proto);
+  unsigned long long int size = graph_proto_copy.ByteSizeLong();
   for (auto input : graph->inputs()) {
     auto name_tensor_pair = initializers.find(input->debugName());
     if (name_tensor_pair == initializers.end()) {
       continue;
     }
-    onnx::GraphProto* graph_proto_copy = new onnx::GraphProto(*graph_proto);
-    auto tensor_proto = graph_proto_copy->add_initializer();
-    const at::Tensor tensor = name_tensor_pair->second;
+    auto tensor_proto = graph_proto_copy.add_initializer();
+    const at::Tensor& tensor = name_tensor_pair->second;
     for (auto d : tensor.sizes()) {
       tensor_proto->add_dims(d);
     }
     tensor_proto->set_data_type(ATenTypeToOnnxType(tensor.scalar_type()));
-    at::Tensor t;
-    if (tensor.is_quantized()) {
-      t = tensor.contiguous();
-    } else {
-      t = tensor.contiguous().cpu();
-    }
-    tensor_proto->set_raw_data(std::string(
-        static_cast<char*>(t.data_ptr()), t.element_size() * t.numel()));
-    sizes += tensor_proto->ByteSizeLong();
-    delete graph_proto_copy;
-    graph_proto_copy = nullptr;
+
+    // Don't actually copy the buffer into tensor_proto since that is expensive.
+    // All we actually need is its size.
+    size += tensor_proto->ByteSizeLong();
```

Comment:
This is fine as is. I'm thinking since this is rather an estimate of GraphProto size, it feels fine to drop `tensor_proto` and only compute size from tensor? And then further drop GraphProto copy on L836?

One can give this a try in a follow-up as well as fix the bug that this function currently neglects constants in the model which could be large. Related #102479, #100873

Reply:
I guess it's unlikely, but maybe these non-weight fields could also push the proto to be over 2GB.